src/chainstate/stacks/db/mod.rs

```diff
@@ -1001,6 +1030,154 @@ impl StacksChainState {
+                                    let coeff = Value::UInt(entry.coeff.into());
+                                    let nonalpha_discount =
+                                        Value::UInt(entry.nonalpha_discount.into());
+                                    let no_vowel_discount =
+                                        Value::UInt(entry.no_vowel_discount.into());
+                                    let buckets: Vec<_> = entry
+                                        .buckets
+                                        .split(";")
+                                        .map(|e| Value::UInt(e.parse::<u64>().unwrap().into()))
+                                        .collect();
+
+                                    TupleData::from_data(vec![
+                                        ("buckets".into(), Value::list_from(buckets).unwrap()),
+                                        ("base".into(), base),
+                                        ("coeff".into(), coeff),
+                                        ("nonalpha-discount".into(), nonalpha_discount),
+                                        ("no-vowel-discount".into(), no_vowel_discount),
+                                    ])
+                                    .unwrap()
+                                };
+
+                                let namespace_props = Value::Tuple(
+                                    TupleData::from_data(vec![
+                                        ("revealed-at".into(), revealed_at),
+                                        ("launched-at".into(), Value::some(launched_at).unwrap()),
+                                        ("lifetime".into(), lifetime),
+                                        ("namespace-import".into(), importer),
+                                        ("price-function".into(), Value::Tuple(price_function)),
+                                    ])
+                                    .unwrap(),
+                                );
+
+                                db.insert_entry(
+                                    &bns_contract_id,
+                                    "namespaces",
+                                    namespace,
+                                    namespace_props,
+                                )?;
+                            }
+                            Ok(())
+                        })
+                        .unwrap();
+                });
+            }
+
+            if let Some(get_names) = boot_data.get_bulk_initial_names.take() {
+                info!("Initializing chain with names");
+                clarity_tx.connection().as_transaction(|clarity| {
+                    clarity
+                        .with_clarity_db(|db| {
+                            let initial_names = get_names();
+                            for entry in initial_names {
+                                let components: Vec<_> =
+                                    entry.fully_qualified_name.split(".").collect();
```

Comment:
Just for safety, you may want to assert that `components` has length 2, and that both `components[0]` and `components[1]` use only the permitted name characters.

Reply:
Checks added in https://github.com/blockstack/stacks-blockchain/pull/2204/commits/52579d5dd4d88987a9dbc53543c8bb85ae971cbc